packages/playground/src/weblets/profile_manager.vue

```diff
@@ -906,6 +904,20 @@ async function getEmail() {
   email.value = await readEmail(grid!);
   loadEmail.value = false;
 }
+async function hasTwinId() {
+  try {
+    await getGrid({ mnemonic: mnemonic.value, keypairType: keypairType.value });
+    await getEmail();
+  } catch (e) {
+    if (e instanceof TwinNotExistError) {
+      isNonActiveMnemonic.value = true;
+    } else {
+      return {
+        message: normalizeError(e, "Something went wrong. please try again."),
+      };
+    }
+  }
+}
```

Comment:
I suggest to refactor the `readEmail` instead to have try and catch.
```ts 
async function getEmail() {
  loadEmail.value = true;
  try {
    const grid = await getGrid({ mnemonic: mnemonic.value, keypairType: keypairType.value });
    if (grid) email.value = await readEmail(grid);
  } catch (e) {
    if (!(e instanceof TwinNotExistError)) {
      return {
        message: normalizeError(e, "Something went wrong. please try again."),
      };
    }
  } finally {
    loadEmail.value = false;
  }
}
```
 The idea is we are already handling the hasTwinId logic in storeAndLogin function;
 If we could just add add try and catch to the get email function, it will be better, i think;
what do you think about this ?